backend/src/test/java/io/github/k_gregory/registry/service/SpringDataUserServiceTest.java

```diff
@@ -0,0 +1,94 @@
+package io.github.k_gregory.registry.service;
+
+import io.github.k_gregory.registry.AbstractIntegrationTest;
+import io.github.k_gregory.registry.model.security.AppUser;
+import io.github.k_gregory.registry.model.security.Authority;
+import io.github.k_gregory.registry.model.security.Group;
+import io.github.k_gregory.registry.repository.UserRepository;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.persistence.EntityManager;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.security.core.authority.AuthorityUtils.createAuthorityList;
+
+
+public class SpringDataUserServiceTest extends AbstractIntegrationTest {
+    @Autowired
+    private EntityManager em;
+
+    @Autowired
+    private SpringDataUserService userService;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Test
+    @Transactional
+    public void getUserAuthorities() {
```

Comment:
Could we be more specific about the purpose of this test? We dhould not fear that the test names will be long - that's a good practice.

Reply:
It tests method named 'getUserAuthorities' in 'SpringDataUserService' class. This method takes `AppUser`, fetches all user permission from DB and all permissions of groups to which user belongs. In this test I create 4 permissions, 3 groups and 4 users. Then I assign some rights to these users and add them to some groups. Then I compare set of permissions returned by `getUserAuthorities` to set of permissions which is correct. 
Which name should I use and would you suggest me to split this method?

Reply:
@k-gregory I would suggest naming that something like
public void getUserAuthorities_WhenAddingAuthoritiesToUsersAndGroups_ReturnsAllAuthorities

That's if I understood the case correctly